scintillometry/channelize/base.py

```diff
@@ -0,0 +1,99 @@
+# Licensed under the GPLv3 - see LICENSE
+
+import operator
+from astropy.utils import lazyproperty
+
+from ..base.base import ModuleBase
+from ..fourier import get_fft_maker
+
+
+__all__ = ['ChannelizeModule']
+
+
+class ChannelizeCore(object):
+
+    # NOTE: maybe force users to pass their own FFT.
+    # NOTE: always assumes time axis is 1.  Probably not a problem.
+    def __init__(self, sample_shape, dtype, nchan, samples_per_block=1,
+                 FFT=None):
+        self.nchan = operator.index(nchan)
+        self.samples_per_block = max(1, operator.index(samples_per_block))
+        FFT = get_fft_maker('numpy') if FFT is None else FFT
+        self._input_shape = (self.samples_per_block, self.nchan) + sample_shape
```

Comment:
I'm hoping to reserve `sample_shape` for the output shape of any class, meaning here it would be the FFT output shape, rather than the input.

It is true that we don't really need `samples_per_block` anymore since the input shape now has it.

Reply:
Yes, that makes sense: we need `sample_shape` for the output.